### PR TITLE
[#88447758] First pass at downloading ICS files for events

### PR DESCRIPTION
**Pull request by netikular**
https://www.pivotaltracker.com/story/show/88447758

@grayden 

This is a first pass at calendar objects for events.  Can you let me know what you think?  I've not started google calendar integration nor is there a pop up to select the kind of integration you want.  I just wanted to prove that I could make an ics file :smile: 

I'm wondering if the development effort to integrate with google calendar will be worth it.  I've made a comment in the ticket regarding it.

Documentation for integrating with google calendar: https://developers.google.com/google-apps/calendar/firstapp

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -59,6 +59,21 @@ class Event
     end
   end
 
+  def to_ics
+    e = Icalendar::Event.new
+    if all_day?
+      e.dtstart     = Icalendar::Values::Date.new(self.datetime.strftime("%Y%m%d"))
+      e.dtend       = Icalendar::Values::Date.new(self.datetime.strftime("%Y%m%d"))
+    else
+      e.dtstart     = Icalendar::Values::DateTime.new(self.datetime.strftime("%Y%m%dT%H%M%S"))
```
Comment (grayden) [inline app/models/event.rb]: @netikular The ical object wants the datetime in UTC? It converts it itself? If so, you'll want to use get_assumed_time which gets the UTC representation of the datetime factoring in how recurring and tv show events should be adjusted.

**Comment by grayden** [comment]
The internet is wonderful.

**Comment by grayden** [comment]
@netikular Am I supposed to merge this? Your pull request says master, your ticket says "not yet".

**Comment by netikular** [comment]
@grayden Don't merge it yet.

What do you think of the implementation?

**Comment by grayden** [comment]
It's great. I can see a service class evolving around making iCal and google calendar invites in the future, but the code is so small for now that I think we have what we need. Again, the datetime that is provided to the iCal event will probably be wrong for recurring and tv show events.

**Comment by netikular** [comment]
TV show events bug me - on a fundamental level :smile: 

I think there needs to be more testing,  Can you put together a list of things we should try out?

When I was testing the event ended up on the right time as it is implemented here.  I'll use the other date time.

**Comment by grayden** [comment]
Since we need to get a time out in UTC we literally just need to try:

* Making all day event calendar items
* Making relative event calendar items
* Making recurring event calendar items
* Making tv show event calendar items

However, the API may have been smart enough to set the value of the event's datetime field to already be the UTC time of the event, so maybe this is a no brainer. Otherwise, the safe way to get the UTC datetime is to call up the user's timezone and use the get_utc_datetime method.

**Pull request merged**